File: String_Functions/generate_all_subsequences/gen_substr.py
# Python Code Example
def helper(s, n, i, sub, container):
    # Base case: when we reach the end of the string
    if i >= n:
        container.append(sub)
        return
    
    # Include the current character in the subsequence
    helper(s, n, i + 1, sub + s[i], container)
    
    # Exclude the current character from the subsequence
    helper(s, n, i + 1, sub, container)

def func_body(s):
    n = len(s)
    result = []
    helper(s, n, 0, "", result)
    return result

if __name__ == "__main__":
    # Main function where the function is called and use case is demonstrated
    s = "abcd"  # Example input
    all_subsequences = func_body(s)

    # Print all subsequences
    for subsequence in all_subsequences:
        print(subsequence)

File: String_Functions/generate_all_substrings/gen_substr.cpp
// C++ Code Example
#include <iostream>
using namespace std;

void func_body(string &s) {
    // The required function which will be added
    int n = s.size();
    for(int i = 0; i<n; i++){
        for(int j = i; j<n; j++){
            cout<<s.substr(i, j-i+1)<<"\n";
        }
    }
}

int main() {
    // Main function where the function is called and use case is demonstrated
    string s = "abcd";
    func_body(s);
    return 0;
}